### PR TITLE
[20.01] Place version output file in a writable directory for containerized jobs.

### PR DESCRIPTION
**Pull request by jmchilton**

### CODE REVIEW

File: lib/galaxy/jobs/__init__.py [hunk 1]
```diff
@@ -1034,11 +1034,11 @@ def validate_outputs(self):
...
 
     def get_version_string_path(self):
-        return os.path.abspath(os.path.join(self.working_directory, COMMAND_VERSION_FILENAME))
+        return os.path.abspath(os.path.join(self.working_directory, "outputs", COMMAND_VERSION_FILENAME))
 
-    # TODO: Remove in Galaxy 20.XX, for running jobs at GX upgrade
+    # TODO: Remove in Galaxy 21.XX, for running jobs at GX upgrade
     def get_version_string_path_legacy(self):
-        return os.path.abspath(os.path.join(self.app.config.new_file_path, "GALAXY_VERSION_STRING_%s" % self.job_id))
+        return os.path.abspath(os.path.join(self.working_directory, COMMAND_VERSION_FILENAME))
```
Comment (mvdbeek) [inline lib/galaxy/jobs/__init__.py:1041]: I think this breaks the unit tests:
```
=================================== FAILURES ===================================
_____________________ JobWrapperTestCase.test_version_path _____________________

self = <unit.jobs.test_job_wrapper.JobWrapperTestCase testMethod=test_version_path>

    def test_version_path(self):
        wrapper = self._wrapper()
        version_path = wrapper.get_version_string_path_legacy()
        expected_path = os.path.join(self.test_directory, "new_files", "GALAXY_VERSION_STRING_345")
>       self.assertEqual(version_path, expected_path)
E       AssertionError: '/tmp/tmpBoY7by/working/COMMAND_VERSION' != '/tmp/tmpBoY7by/new_files/GALAXY_VERSION_STRING_345'

test/unit/jobs/test_job_wrapper.py:57: AssertionError
----------------------------- Captured stderr call -----------------------------
DEBUG:galaxy.jobs:(345) Working directory for job is: /tmp/tmpBoY7by/working
------------------------------ Captured log call -------------------------------
DEBUG    galaxy.jobs:__init__.py:1117 (345) Working directory for job is: /tmp/tmpBoY7by/working
____________________ TaskWrapperTestCase.test_version_path _____________________

self = <unit.jobs.test_job_wrapper.TaskWrapperTestCase testMethod=test_version_path>

    def test_version_path(self):
        wrapper = self._wrapper()
        version_path = wrapper.get_version_string_path_legacy()
        expected_path = os.path.join(self.test_directory, "new_files", "GALAXY_VERSION_STRING_345")
>       self.assertEqual(version_path, expected_path)
E       AssertionError: '/tmp/tmpufPE6E/working/COMMAND_VERSION' != '/tmp/tmpufPE6E/new_files/GALAXY_VERSION_STRING_345'

test/unit/jobs/test_job_wrapper.py:57: AssertionError
----------------------------- Captured stderr call -----------------------------
DEBUG:galaxy.jobs:(345) Working directory for job is: /tmp/tmpufPE6E/working
------------------------------ Captured log call -------------------------------
DEBUG    galaxy.jobs:__init__.py:1117 (345) Working directory for job is: /tmp/tmpufPE6E/working
```

**Comment by jmchilton** [comment]
Ugh - thanks for re-milestoning this to 20.01 and sorry for letting this slip. This is a bug fix and belongs there.

**Comment by jmchilton** [comment]
GUI seems to not match what my git thinks is in the branch 😿 - maybe changing base and force pushing too close together broke something?

**Review by mvdbeek — APPROVED** [review]
Awesome, this looks good to me, take off the WIP ?

**Pull request merged**